src/main/java/htsjdk/variant/variantcontext/VariantContextBuilder.java

```diff
@@ -323,12 +323,26 @@ public VariantContextBuilder rmAttributes(final List<String> keys) {
      * collection, so methods that want to add / remove records require the attributes to be copied to a
      */
     private void makeAttributesModifiable() {
-        if ( ! attributesCanBeModified ) {
+        if ( !attributesCanBeModified) {
             this.attributesCanBeModified = true;
             if (attributes == null) {
-            	this.attributes = new HashMap<String, Object>();
+                this.attributes = new HashMap<>();
             } else {
-            	this.attributes = new HashMap<String, Object>(attributes);
+                this.attributes = new HashMap<>(attributes);
+            }
+        }
+    }
+
+    /**
+     * Makes the filters modifiable.
+     */
+    private void makeFiltersModifiable() {
+        if ( !filtersCanBeModified) {
+            this.filtersCanBeModified = true;
+            if (filters == null) {
```

Comment:
could be written as
```
this.filters = new HashSet<>();
if (filters != null) {
    this.filters.addAll(filters);
}
```

Reply:
hmmm. that will not work as written, since filters will never be null.

Reply:
but I get your drift.